src/LanguageServer/Impl/Implementation/Server.Editor.cs

```diff
@@ -112,5 +112,23 @@ await InvokeExtensionsAsync(async (ext, token)
             _log?.Log(TraceEventType.Verbose, $"Rename in {uri} at {@params.position}");
             return new RenameSource(Services).RenameAsync(uri, @params.position, @params.newName, cancellationToken);
         }
+
+        public async Task<CodeAction[]> CodeAction(CodeActionParams @params, CancellationToken cancellationToken) {
+            var uri = @params.textDocument.uri;
+            _log?.Log(TraceEventType.Verbose, $"Code Action in {uri} at {@params.range}");
+
+            if (@params.context.diagnostics?.Length == 0) {
+                return Array.Empty<CodeAction>();
+            }
+
+            // * NOTE * CodeActionParams.CodeActionContext._version is always null. PublishDiagnostic doesn't send version
```

Comment:
Safe to remove `_version`. We've been cleaning up old unused custom fields that PTVS used to exchange with old LS.